src/main/java/chess/piece/type/Piece.java

```diff
@@ -0,0 +1,84 @@
+package chess.piece.type;
+
+import java.util.Map;
+import java.util.Objects;
+
+import chess.board.Location;
+import chess.score.Score;
+import chess.team.Team;
+
+public abstract class Piece {
+    private static final char BLACk_KING_VALUE = 'K';
+    private static final char WHITE_KING_VALUE = 'k';
+
+    public final char name;
+    protected final Score score;
+
+    Piece(char name, Score score) {
+        this.name = name;
+        this.score = score;
+    }
+
+    public abstract boolean canMove(Location now, Location after);
+
+    public boolean isSameTeam(Team team) {
+        return isBlack() == team.isBlack();
+    }
+
+    public boolean isSameTeam(boolean black) {
+        return isBlack() == black;
+    }
+
+    public boolean isSameTeam(Piece piece) {
+        return isBlack() == piece.isBlack();
+    }
+
+    public boolean isKing() {
+        return this.name == BLACk_KING_VALUE || this.name == WHITE_KING_VALUE;
+    }
+
+    protected boolean isBlack() {
+        return Character.isUpperCase(name);
+    }
+
+    public char getName() {
+        return name;
+    }
+
+    public Score getScore() {
+        return score;
+    }
+
+    public boolean hasObstacle(Map<Location, Piece> board, Location now, Location destination) {
+        Location nowLocation;
+        for (int weight = 1;; weight++) {
+            nowLocation = now.calculateNextLocation(destination, weight);
+            if (nowLocation.equals(destination)) {
+                break;
```

Comment:
```suggestion
                return false;
```

코드를 읽는 입장에서 해당 위치에 break가 있다면 메서드의 마지막 부분을 확인해야 하는데요.
지금은 메서드가 짧아 파악이 쉽지만 코드가 복잡하다면 코드를 분석해야 하는 일이 발생할 수 있습니다.
이러한 경우 바로 return하여 코드를 읽지 않도록 하는 것도 가독성을 높이는 방법이 될 수 있습니다.

Reply:
`break문 사용할 때 바로 return하는 방향을 고려하라`
 - break가 있으면 메서드의 마지막 부분을 확인해야한다. 메서드가 짧으면 문제가 없지만 코드가 복잡하다면 코드를 분석해야하는 일이 발생할 수 있다.
 - 그러므로 바로 return하여 코드를 읽지 않도록 하여 가독성을 높이자.